apps/src/templates/projects/submitProjectDialog/submitProjectApi.ts

```diff
@@ -0,0 +1,28 @@
+import {SubmissionStatusType} from '@cdo/apps/lab2/views/dialogs/types';
+import HttpClient from '@cdo/apps/util/HttpClient';
+
+interface SubmissionStatusResponse {
+  status: SubmissionStatusType;
+}
+
+/**
+ * TODO: Sends a post request to submit the project.
+ */
+export async function submitProject(submissionDescription: string) {
+  console.log('submitProject', submissionDescription);
+}
+
+/**
+ * Sends a get request to submit the project.
+ */
+export async function getSubmissionStatus(): Promise<
+  SubmissionStatusResponse | undefined
+> {
+  try {
+    const response = await HttpClient.fetchJson(`submission_status`);
+    return response.value as SubmissionStatusResponse;
+  } catch (error) {
+    // A signed out user does not have access to `submission_status`.
+    return undefined;
```

Comment:
What error does the server return exactly if the user is signed out? We should probably differentiate from an intentional server error and an unexpected error that should be caught and logged.

Reply:
They get a 302 from https://github.com/code-dot-org/code-dot-org/blob/70172fb76632b0f2ff639ebbc98bbf0d8022e3ef/dashboard/app/controllers/projects_controller.rb#L5.
Since I'm using `fetchJson`, the error is thrown at https://github.com/code-dot-org/code-dot-org/blob/70172fb76632b0f2ff639ebbc98bbf0d8022e3ef/apps/src/util/HttpClient.ts#L63.
Sure - I can include this server logging in https://codedotorg.atlassian.net/browse/LABS-1196

Reply:
Noting a couple options for a follow-up:
- Add `submission_status` as an exception for `:authentic_user!` at https://github.com/code-dot-org/code-dot-org/blob/f680de6a3dac057f059cc93ed5c5076c4e659d0a/dashboard/app/controllers/projects_controller.rb#L5 and add specific error message similar to https://github.com/code-dot-org/code-dot-org/blob/f680de6a3dac057f059cc93ed5c5076c4e659d0a/dashboard/app/controllers/projects_controller.rb#L343-L344
- Add `submission_status` to `ability.rb` under `user.persisted`.

@bencodeorg - curious if you suggest a preference (this will be done in follow-up).

Reply:
mmm can I say a combination? I don't think it makes sense to authenticate_user! here (b/c like Sanchit said, it generates an HTML response), and I think you should also add submission_status to ability.rb with the gating logic checking that the project is owned by the requester (not just that the user is signed in)? Not sure if you need a nice error message at this point since Sam said we're not showing anything to the user right? Just a 403 so you can not show the UI

Reply:
Oh maybe you do need a special error message so you can differentiate from other 403 types? At that point, might as well just tell the user that they need to sign in to submit :)